### PR TITLE
alan0201tw term proposal

### PR DESCRIPTION
**Pull request by alan0201tw**
I excluded the unwanted files.
Sorry for the trouble.

### CODE REVIEW

File: term_proposal/alan0201tw/proposal.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+### Determined direction
+
+* Animation-oriented, abandon interactivity and focus on producing nice quality videos
+* Have minimal dependencies, this way building it will be easy, and thus much more friendly to novice programmers that are trying to jump in the field of computer graphics
+    * However, I am thinking about using stb_image for image save/load, and ffmpeg for generating videos
+    * stb_image is a single header library, and do not require extra set ups, so it probably won't complicate stuffs
+    * ffmpeg can be used as an external tool, and be used with shell scripts, thus not affecting the build chain, so it should be okay, too
+
+### TBD
```
Comment (yungyuc) [inline term_proposal/alan0201tw/proposal.md]: Please give this section a title.  You probably also want to beef up the contents too?

File: term_proposal/alan0201tw/proposal.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+* Use physics-based shading or not?
+* Is providing surface normal the responsibility of the surface, or should be evaluated by ray-casting?
+
+
+## Implementation
+
+My first instinct is to implement everything in c++, but having a python interface for setting up scenes sounds quite nice, so I might do that (if I have the time).
```
Comment (yungyuc) [inline term_proposal/alan0201tw/proposal.md]: I suggest to use a more formal sentence.  Please also elaborate more about how do you plan to separate the C++ implementation and the Python wrapper.

File: term_proposal/alan0201tw/proposal.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+The first milestone will be just rendering some cubes or spheres on the image, then I can build extra features on top of it. Like lighting, materials, and mesh operations like masking and blending.
```
Comment (yungyuc) [inline term_proposal/alan0201tw/proposal.md:58]: You will also need to discuss the plan for GUI (or not having it).

File: term_proposal/alan0201tw/proposal.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+
```
Comment (yungyuc) [inline term_proposal/alan0201tw/proposal.md:67]: What's your plan of testing?

**Review by yungyuc — CHANGES_REQUESTED** [review]

File: term_proposal/alan0201tw/proposal.md (hunk 1 above)
Comment (alan0201tw) [reply term_proposal/alan0201tw/proposal.md]: I changed the title to 'Considering design choices', I will add other entries once I think of any.

File: term_proposal/alan0201tw/proposal.md (hunk 2 above)
Comment (alan0201tw) [reply term_proposal/alan0201tw/proposal.md]: I have refactored the sentence to multiple sections and remove casual syntaxes.

File: term_proposal/alan0201tw/proposal.md (hunk 3 above)
Comment (alan0201tw) [reply term_proposal/alan0201tw/proposal.md:58]: Related plans added to section 'Implementation'.

File: term_proposal/alan0201tw/proposal.md (hunk 4 above)
Comment (alan0201tw) [reply term_proposal/alan0201tw/proposal.md:67]: Related plans added to section 'Implementation'.

**Review by alan0201tw — COMMENTED** [review]
Thank you for your feedback, I have updated some content base on your comments.
Please check the validity of this proposal.
If any further changes is required, I will resolve it at earliest convenience.

File: term_proposal/alan0201tw/proposal.md [hunk 5]
```diff
@@ -41,18 +41,29 @@ And since I have a working (simple, but working!) ray tracer at hand, I can do s
...
+### Considering design choices
 
 * Use physics-based shading or not?
 * Is providing surface normal the responsibility of the surface, or should be evaluated by ray-casting?
 
 
 ## Implementation
 
-My first instinct is to implement everything in c++, but having a python interface for setting up scenes sounds quite nice, so I might do that (if I have the time).
+The render engine itslef will be implemented in c++. It will be a standalone application. Scene contents will be controlled through c++ code.
```
Comment (yungyuc) [inline term_proposal/alan0201tw/proposal.md]: When you say application, I assume it is an executable, and not a shared object.  If this is true, please make it explicit in the proposal.

I'd like to add that a standalone executable will make it difficult to test and script in Python.  It is OK to make this decision, though, as long as you are fully aware of the consequences.

**Review by yungyuc — APPROVED** [review]

File: term_proposal/alan0201tw/proposal.md (hunk 5 above)
Comment (alan0201tw) [reply term_proposal/alan0201tw/proposal.md]: Thank you for pointing out the inaccurate word use. I did mean 'executable' when I used the word 'application'.

Designing this engine as a standalone executable is based on my previous experience on ray tracers. I will surely start this project as a standalone executable, but I might switch it to the form of shared library once I am comfortable with it.

**Pull request merged**